### DISCUSSION TITLE
Mixcr for long-read data?

### DISCUSSION THREAD

**Discussion by amaraalexan** [milaboratory/mixcr#913] [Q&A] [answered]
Hello, 

I am a new user of mixcr and am trying to analzye enriched BCR targets sequenced with oxford nanopore. Is there anyway to adapt the parameters of a mixcr pipeline to improve efficiency for this type of data? Any advice or recommendations are appreciated as I am a student and have a lot to learn!

**Comment by amikelov** [comment]
Hi amaraalexan,

We are glad to help with this, as we are very interested in supporting long-reads applications in BCR repertoire profiling.
Could you please elaborate a little bit on the structure and nature of your libraries?
1) what is the source of cells and species for your libraries? tissues/blood/cell lines etc
1) what is the starting material for your libraries, DNA or RNA?
2) what kind of enrichment for BCR targets did you use? Is it an amplicon sequencing?
3) what kind of primers/adapters were used. For amplicon sequencing - what receptor regions the primers are specific to (e.g. exon 4 of a C gene and multiplex V gene primers).
4) Are UMIs utilized? Error rates in oxford nanopore might be quite detrimental for the accuracy of the resulting clonesets, but UMIs could help a lot here.

Best regards,
Artem

**Comment by PavithraV0223** [comment]
Greetings, 
I actually have a similar question, I trying to analyse BCR regions using the Oxford Nanopore data and it is from alpaca.  The staring material would be the cDNA, its a long read seq.

**Comment by hsome99** [comment]
Hi, 

I wonder if MIXCR would be able to process paired VH:VL (CJDV-Linker-VJ) Miseq sequencing data? 

<img width="540" alt="image" src="https://github.com/milaboratory/mixcr/assets/137711659/fa27e45a-1a8f-4d6f-8fc4-9058c4fd7391">

**Reply by PoslavskySV** [reply]
Let's assume you have one sample as:
```
Sample_R1_001.fastq.gz
Sample_R2_001.fastq.gz
```
Then you can use the following command to conduct the analysis of your data:
```
mixcr analyze generic-bcr-amplicon \
  --species human \
  --dna \
  --floating-left-alignment-boundary \
  --floating-right-alignment-boundary C \
  --add-step refineTagsAndSort \
  -MrefineTagsAndSort.runCorrection=false \
  -Malign.readIdAsCellTag=true \
  --append-export-clones-field -tags Cell \
  -MexportClones.splitFilesBy=[] \
  Sample_R{{CELLSPLIT:n}}_001.fastq.gz output
```	
	
Meaning of the parameters is the following:

`--dna`	
Assume you have DNA starting material

`--floating-left-alignment-boundary`
Use semiglobal alignment algorithm at 5'

`--floating-right-alignment-boundary C`
Use semiglobal alignment algorithm at 3' (in C region)

`
-Malign.readIdAsCellTag=true
`
`
--add-step refineTagsAndSort
`
`
-MrefineTagsAndSort.runCorrection=false 
`
These options will instruct MiXCR to use readID as "cell" barcode. We also need to add tag processing step to sort alignments by "cell" tag for clonotype assembly. Finally, we turn off any error correction for these "cell" tags.

`--append-export-clones-field -tags Cell`
Add "cell" tag column to export. This column will contain readID.

`-MexportClones.splitFilesBy=[]`
Instruct MiXCR to not split output clonotype table into different chains.

`Sample_R{{CELLSPLIT:n}}_001.fastq.gz output`
Instructs MiXCR to separately process R1 and R2 as two different sequences, but with the same cell barcode.

As result you will have output.tsv file with column 'Cell' containing the cell tag (read ID). There will be two rows corresponding to each "cell" tag - one for IGH and one for IGK.

**Comment by PoslavskySV** [comment, marked as answer]
MiXCR has built-in support for ONT data:

  https://mixcr.com/mixcr/reference/overview-built-in-presets/#oxford-nanopore-technologies
